src/parcsr_mv/par_csr_filter_device.c

```diff
@@ -171,12 +172,20 @@ hypreGPUKernel_ParCSRMatrixBlkFilterFill(hypre_DeviceItem &item,
             B_offd_j[idx] = col;
             B_offd_a[idx] = A_offd_a[j];
 
-#ifndef HYPRE_USING_SYCL
             if (col < A_num_cols_offd)
             {
+#if defined(HYPRE_USING_SYCL)
+               /* WM: question - why is the atomic necessary here? */
+               col_map_marker[col] = 1;
+               /* auto atomic_key = sycl::atomic_ref < */
+               /*                   HYPRE_Int, sycl::memory_order::relaxed, */
+               /*                   sycl::memory_scope::device, */
+               /*                   sycl::access::address_space::local_space > (col_map_marker[col]); */
+               /* atomic_key.fetch_or(1); */
+#else
                atomicOr(col_map_marker + col, 1);
```

Comment:
@victorapm , I think this is code you recently added. I was wondering why you put an atomic operation here, since it seems like you are just writing 1 to the memory location when the if condition is satisfied. Even if you have multiple threads trying to write the same location at once, I don't see why the atomic is necessary?

Reply:
Thanks for catching, Wayne. You are right, we can remove `atomicOr` here